scanpy/_metadata.py

```diff
@@ -42,6 +42,12 @@ def refresh_entry_points():
 
 
 def within_flit():
+    """\
+    Checks if we are being imported by flit.
+    This is necessary to make the above code work while not all dependencies are there.
+    There are a few options to make this unnecessary, see:
+    https://github.com/takluyver/flit/issues/253#issuecomment-737870438
```

Comment:
I think this could be more clear, I'm not sure what "the above code" means.

The specific issue is `__version__`, right? And that it's important to be able to get the version number without dependencies being installed (could you include why that is as well)? It's probably also good to mention that this is a hack.